src/main/java/model/Operation.java

```diff
@@ -0,0 +1,27 @@
+package model;
+
+public enum Operation {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Operation getOperationFromCode(String code) {
+        for (Operation operation : Operation.values()) {
+            if (operation.getCode().equals(code)) {
+                return operation;
+            }
+        }
+        throw new RuntimeException("Invalid operation type " + code);
```

Comment:
The exception message could be more informative by suggesting the correct code format or expected values.